### PR TITLE
Support Named Pipe for DAP Traffics on Windows

### PR DESCRIPTION
**Pull request by kuchungmsft**
Support Named Pipe for DAP Traffics on Windows

- Part of #1730304
- Include unit test.

image
image

Thanks for your interest in contributing to CMake!  The GitHub repository
is a mirror provided for convenience, but CMake does not use GitHub pull
requests for contribution.  Please see

  https://gitlab.kitware.com/cmake/cmake/-/tree/master/CONTRIBUTING.rst

for contribution instructions.  GitHub OAuth may be used to sign in.

### CODE REVIEW

File: Source/cmDebuggerNamedPipeWin32.cxx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+   file Copyright.txt or https://cmake.org/licensing for details.  */
+
+#include "cmsys/Encoding.hxx"
+
+#include "cmDebuggerNamedPipe.h"
+#include "cmSystemTools.h"
+
+namespace cmDebugger {
+
+cmDebuggerNamedPipe::cmDebuggerNamedPipe(std::string const& name)
```
Comment (yuehuang010) [inline Source/cmDebuggerNamedPipeWin32.cxx]: Is this constructor used?

Comment (kuchungmsft) [reply Source/cmDebuggerNamedPipeWin32.cxx]: Yes, used in cmDebuggerAdapterFactory.cxx.

image

**Review by bobbrow — APPROVED** [review]

File: Source/cmDebuggerNamedPipeWin32.cxx (hunk 1 above)
Comment (kuchungmsft) [reply Source/cmDebuggerNamedPipeWin32.cxx]: @yuehuang010 I've refactored the code a bit, got rid of the cmDebuggerNamedPipe.h and hide the platform-specific details in the implementation files as much as possible, please take another look, thanks.

**Comment by yuehuang010** [comment]
Thanks, it looks better.

**Pull request merged**